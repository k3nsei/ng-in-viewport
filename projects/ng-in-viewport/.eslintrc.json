{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "rules": {
        "@typescript-eslint/no-explicit-any": "off",
        "@typescript-eslint/no-non-null-assertion": "off",
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "inViewport",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "in-viewport",
            "style": "kebab-case"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "rules": {}
    }
  ]
}
