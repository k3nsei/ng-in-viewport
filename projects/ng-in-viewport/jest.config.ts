import type { Config } from 'jest';

const jestConfig: Config = {
  displayName: 'ng-in-viewport',
  preset: 'jest-preset-angular',
  // globalSetup: 'jest-preset-angular/global-setup',
  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
  coverageDirectory: '<rootDir>/../../coverage/ng-in-viewport',
  coverageProvider: 'v8',
  coverageReporters: ['lcovonly', 'text', 'html-spa'],
  coverageThreshold: {
    global: {
      branches: 95,
      functions: 95,
      lines: 90,
      statements: 90,
    },
  },
  coveragePathIgnorePatterns: ['node_modules/', 'enums/', 'index.ts', 'public-api.ts', 'in-viewport.module.ts'],
};

export default jestConfig;
