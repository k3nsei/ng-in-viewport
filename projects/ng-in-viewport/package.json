{
  "name": "ng-in-viewport",
  "version": "15.0.2",
  "description": "Allows us to check if an element is within the browsers visual viewport",
  "homepage": "https://k3nsei.github.io/ng-in-viewport/",
  "bugs": {
    "url": "https://github.com/k3nsei/ng-in-viewport/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/k3nsei/ng-in-viewport.git"
  },
  "license": "MIT",
  "author": {
    "name": "Piotr StÄ™pniewski",
    "email": "k3nsei.pl@gmail.com",
    "url": "https://github.com/k3nsei"
  },
  "dependencies": {
    "lodash-es": "^4.17.21",
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^14.0.0 || >=15.0.0",
    "@angular/core": "^14.0.0 || >=15.0.0"
  },
  "publishConfig": {
    "provenance": true
  }
}
