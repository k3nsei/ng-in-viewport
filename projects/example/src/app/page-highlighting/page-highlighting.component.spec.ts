import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';
import { MockDirective } from 'ng-mocks';

import { InViewportAction, InViewportDirective } from 'ng-in-viewport';

import { PageHighlightingComponent } from './page-highlighting.component';

describe('GIVEN PageHighlightingComponent', () => {
  let spectator: Spectator<PageHighlightingComponent>;
  let component: PageHighlightingComponent;
  const createComponent = createComponentFactory({
    component: PageHighlightingComponent,
    overrideComponents: [
      [
        PageHighlightingComponent,
        {
          remove: { imports: [InViewportDirective] },
          add: { imports: [MockDirective(InViewportDirective)] },
        },
      ],
    ],
  });

  describe('WHEN component was created', () => {
    beforeEach(() => {
      spectator = createComponent();
      component = spectator.component;
    });

    it('THEN instance should exists', () => {
      expect(component).toBeTruthy();
    });

    it('THEN host should match snapshot', () => {
      const hostElement = spectator.element;

      expect(hostElement).toMatchSnapshot();
    });

    it('THEN host should render 100 inactive tiles', () => {
      const tiles = spectator.queryAll('.grid-tile.inactive');

      expect(tiles.length).toBe(100);
    });

    describe('AND `highlight` method was called', () => {
      beforeEach(() => {
        const tile = spectator.query('.grid-tile.inactive');

        component.highlight({ target: tile, visible: true } as InViewportAction);
      });

      it('THEN one tile should be active', () => {
        const tile = spectator.query('.grid-tile.active');

        expect(tile).toBeTruthy();
      });
    });
  });
});
