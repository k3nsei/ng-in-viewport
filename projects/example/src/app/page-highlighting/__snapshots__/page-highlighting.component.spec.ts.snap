// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN PageHighlightingComponent WHEN component was created THEN host should match snapshot 1`] = `
<div
  id="root1"
>
  <mat-grid-list
    class="mat-grid-list"
    cols="2"
    rowheight="2:1"
  >
    <div>
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px; margin-top: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           1 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="margin-top: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           2 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           3 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           4 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           5 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           6 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           7 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           8 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           9 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           10 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           11 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           12 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           13 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           14 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           15 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           16 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           17 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           18 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           19 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           20 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           21 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           22 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           23 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           24 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           25 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           26 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           27 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           28 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           29 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           30 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           31 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           32 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           33 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           34 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           35 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           36 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           37 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           38 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           39 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           40 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           41 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           42 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           43 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           44 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           45 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           46 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           47 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           48 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           49 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           50 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           51 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           52 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           53 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           54 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           55 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           56 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           57 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           58 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           59 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           60 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           61 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           62 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           63 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           64 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           65 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           66 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           67 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           68 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           69 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           70 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           71 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           72 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           73 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           74 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           75 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           76 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           77 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           78 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           79 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           80 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           81 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           82 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           83 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           84 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           85 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           86 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           87 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           88 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           89 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           90 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           91 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           92 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           93 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           94 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           95 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           96 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           97 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           98 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
        style="left: 0px;"
      >
        <div
          class="mat-grid-tile-content"
        >
           99 
        </div>
      </mat-grid-tile>
      
      <mat-grid-tile
        class="mat-grid-tile grid-tile inactive"
        colspan="1"
        inviewport=""
        rowspan="1"
      >
        <div
          class="mat-grid-tile-content"
        >
           100 
        </div>
      </mat-grid-tile>
      
      
    </div>
  </mat-grid-list>
</div>
`;
